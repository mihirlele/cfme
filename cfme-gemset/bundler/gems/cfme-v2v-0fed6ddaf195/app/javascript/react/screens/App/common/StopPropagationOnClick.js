import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Prevents a click on an inner element (e.g. a DropdownKebab) from triggering a click on an ancestor (e.g. a table row).
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const StopPropagationOnClick = ({ children }) => (
  <span onClick={e => e.stopPropagation()} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

StopPropagationOnClick.propTypes = {
  children: PropTypes.node
};

export default StopPropagationOnClick;
