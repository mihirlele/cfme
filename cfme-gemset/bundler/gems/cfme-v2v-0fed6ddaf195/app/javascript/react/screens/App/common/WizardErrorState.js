import React from 'react';
import PropTypes from 'prop-types';

const WizardErrorState = ({ title, message, onClose }) => (
  <div className="wizard-pf-complete blank-slate-pf">
    <div className="wizard-pf-success-icon">
      <span className="pficon pficon-error-circle-o" />
    </div>
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
    {onClose ? (
      <button type="button" className="btn btn-lg btn-primary" onClick={onClose}>
        {__('Close')}
      </button>
    ) : null}
  </div>
);

WizardErrorState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
  onClose: PropTypes.func
};

WizardErrorState.defaultProps = {
  message: ''
};

export default WizardErrorState;
