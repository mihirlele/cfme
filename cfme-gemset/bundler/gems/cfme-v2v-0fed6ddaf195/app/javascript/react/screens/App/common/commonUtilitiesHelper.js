const commonUtilitiesHelper = {
  getMostRecentEntityByCreationDate: entities =>
    entities && entities.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current)),

  groupBy: (items, key) =>
    items &&
    items.reduce(
      (result, item) => ({
        ...result,
        [item[key]]: [...(result[item[key]] || []), item]
      }),
      {}
    ),

  flatten: (arrayItems, flattenedArray) =>
    arrayItems && arrayItems.map(task => task.map(arrayElement => flattenedArray.push(arrayElement)))
};

export default commonUtilitiesHelper;
