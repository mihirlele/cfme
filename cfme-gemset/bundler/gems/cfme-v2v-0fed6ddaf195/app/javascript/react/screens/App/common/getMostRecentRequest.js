const getMostRecentRequest = requests =>
  requests.length && requests.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current));

export default getMostRecentRequest;
