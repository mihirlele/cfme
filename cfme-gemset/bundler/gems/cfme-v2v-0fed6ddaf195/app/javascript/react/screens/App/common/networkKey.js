/**
 * returns a unique network based on current UI model
 */
export const networkKey = network => `${network.clusterId}\\${network.uid_ems || network.ems_ref}`;
