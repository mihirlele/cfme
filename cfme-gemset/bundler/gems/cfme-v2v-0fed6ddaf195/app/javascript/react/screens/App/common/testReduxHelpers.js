import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import configureMockStore from 'redux-mock-store';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));

export const mockStore = initialState => configureMockStore(middlewares)(initialState);
