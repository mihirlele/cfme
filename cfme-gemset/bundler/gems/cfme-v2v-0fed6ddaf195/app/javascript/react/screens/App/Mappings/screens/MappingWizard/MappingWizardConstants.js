import { OPENSTACK, RHV, V2V_TARGET_PROVIDERS } from '../../../../../../common/constants';

export const V2V_SET_TRANSFORMATIONS_BODY = 'V2V_SET_TRANSFORMATIONS_BODY';
export const V2V_SHOW_WARNING_MODAL = 'V2V_SHOW_WARNING_MODAL';
export const V2V_HIDE_WARNING_MODAL = 'V2V_HIDE_WARNING_MODAL';
export const V2V_SHOW_ALERT = 'V2V_SHOW_ALERT';
export const V2V_HIDE_ALERT = 'V2V_HIDE_ALERT';
export const V2V_MAPPING_WIZARD_EXITED = 'V2V_MAPPING_WIZARD_EXITED';
export const V2V_SHOW_MAPPING_WIZARD = 'V2V_SHOW_MAPPING_WIZARD';
export const V2V_HIDE_MAPPING_WIZARD = 'V2V_HIDE_MAPPING_WIZARD';
export { OPENSTACK, RHV, V2V_TARGET_PROVIDERS };

export const V2V_TARGET_PROVIDER_STORAGE_KEYS = {
  [RHV]: 'storages',
  [OPENSTACK]: 'cloud_volume_types'
};

export const V2V_TARGET_PROVIDER_NETWORK_KEYS = { [RHV]: 'lans', [OPENSTACK]: 'cloud_networks' };

export const TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES = {
  cluster: 'EmsCluster',
  datastore: 'Storage',
  network: 'Lan'
};

export const TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES = {
  [OPENSTACK]: {
    cluster: 'CloudTenant',
    datastore: 'CloudVolumeType',
    network: 'CloudNetwork'
  },
  [RHV]: {
    cluster: 'EmsCluster',
    datastore: 'Storage',
    network: 'Lan'
  }
};
