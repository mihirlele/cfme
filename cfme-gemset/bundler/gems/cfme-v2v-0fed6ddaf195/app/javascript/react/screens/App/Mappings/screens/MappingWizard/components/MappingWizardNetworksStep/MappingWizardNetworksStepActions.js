import URI from 'urijs';
import API from '../../../../../../../../common/API';
import {
  FETCH_V2V_SOURCE_NETWORKS,
  FETCH_V2V_TARGET_NETWORKS,
  FETCH_V2V_PUBLIC_CLOUD_NETWORKS,
  QUERY_ATTRIBUTES
} from './MappingWizardNetworksStepConstants';
import { V2V_TARGET_PROVIDER_NETWORK_KEYS } from '../../MappingWizardConstants';

const _filterSourceNetworks = response => {
  const { data } = response;
  if (data.lans) {
    const sourceNetworks = data.lans.map(lan => ({
      ...lan,
      providerName: data.ext_management_system.name,
      clusterId: data.id
    }));
    return {
      sourceNetworks
    };
  }
  return {
    sourceNetworks: []
  };
};

const _getSourceNetworksActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_SOURCE_NETWORKS,
    payload: new Promise((resolve, reject) => {
      API.get(url)
        .then(response => {
          resolve(_filterSourceNetworks(response));
        })
        .catch(e => {
          reject(e);
        });
    })
  });

export const fetchSourceNetworksAction = (url, id) => {
  const uri = new URI(`${url}/${id}`);
  // creates url like: http://localhost:3000/api/clusters/1?attributes=lans
  uri.addSearch({ attributes: QUERY_ATTRIBUTES.source });

  return _getSourceNetworksActionCreator(uri.toString());
};

const _filterTargetNetworks = (response, targetProvider) => {
  const { data } = response;
  if (data[V2V_TARGET_PROVIDER_NETWORK_KEYS[targetProvider]]) {
    const targetNetworks = data[V2V_TARGET_PROVIDER_NETWORK_KEYS[targetProvider]].map(network => ({
      ...network,
      providerName: data.ext_management_system.name,
      clusterId: data.id
    }));

    return {
      targetNetworks
    };
  }
  return {
    targetNetworks: []
  };
};

const _getTargetNetworksActionCreator = (url, targetProvider) => dispatch =>
  dispatch({
    type: FETCH_V2V_TARGET_NETWORKS,
    payload: new Promise((resolve, reject) => {
      API.get(url)
        .then(response => {
          resolve(_filterTargetNetworks(response, targetProvider));
        })
        .catch(e => {
          reject(e);
        });
    })
  });

export const fetchTargetNetworksAction = (url, id, targetProvider) => {
  const uri = new URI(`${url}/${id}`);
  // creates url like: http://localhost:3000/api/clusters/1?attributes=lans
  uri.addSearch({ attributes: QUERY_ATTRIBUTES[targetProvider] });

  return _getTargetNetworksActionCreator(uri.toString(), targetProvider);
};

const _normalizePublicCloudNetworks = (response, targetCloudTenant) => {
  const {
    data: { resources: networks }
  } = response;

  return networks.map(network => ({
    ...network,
    providerName: targetCloudTenant.ext_management_system.name,
    clusterId: targetCloudTenant.id
  }));
};

const _getPublicCloudNetworksActionCreator = (url, targetCloudTenant) => dispatch =>
  dispatch({
    type: FETCH_V2V_PUBLIC_CLOUD_NETWORKS,
    payload: new Promise((resolve, reject) => {
      API.get(url)
        .then(response => {
          resolve(_normalizePublicCloudNetworks(response, targetCloudTenant));
        })
        .catch(e => reject(e));
    })
  });

export const fetchPublicCloudNetworksAction = (url, targetCloudTenant) => {
  const uri = new URI(`${url}/${targetCloudTenant.ext_management_system.id}/cloud_networks`);
  uri.addSearch({ expand: 'resources' });

  return _getPublicCloudNetworksActionCreator(uri.toString(), targetCloudTenant);
};
