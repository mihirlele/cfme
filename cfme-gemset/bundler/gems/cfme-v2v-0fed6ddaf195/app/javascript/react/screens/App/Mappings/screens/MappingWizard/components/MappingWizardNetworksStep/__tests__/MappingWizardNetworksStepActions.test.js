import { mockStore } from '../../../../../../common/testReduxHelpers';
import { mockRequest, mockReset } from '../../../../../../../../../common/mockRequests';
import { fetchTargetNetworksAction, fetchPublicCloudNetworksAction } from '../MappingWizardNetworksStepActions';
import {
  targetClusters,
  cloudTenants
} from '../../../../../../../../../redux/common/targetResources/targetResources.fixtures';
import { V2V_TARGET_PROVIDER_NETWORK_KEYS, V2V_TARGET_PROVIDERS } from '../../../MappingWizardConstants';

const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('fetchTargetNetworksAction', () => {
  describe('if the target provider is OSP,', () => {
    const targetProvider = V2V_TARGET_PROVIDERS[1].id;
    const [targetCloudTenant] = cloudTenants.resources;

    test('dispatches PENDING and FULFILLED actions and sets the clusterId and providerName', () => {
      mockRequest({
        method: 'GET',
        url: '/api',
        status: 200,
        response: {
          data: {
            ...targetCloudTenant,
            ext_management_system: { name: 'some provider' },
            [V2V_TARGET_PROVIDER_NETWORK_KEYS[targetProvider]]: [{}]
          }
        }
      });

      return store.dispatch(fetchTargetNetworksAction('/api', '1', targetProvider)).then(() => {
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
      });
    });
  });

  describe('if the target provider is RHV,', () => {
    const targetProvider = V2V_TARGET_PROVIDERS[0].id;
    const [targetCluster] = targetClusters.resources;

    test('dispatches PENDING and FULFILLED actions and sets the clusterId and providerName', () => {
      mockRequest({
        method: 'GET',
        url: '/api',
        status: 200,
        response: {
          data: {
            ...targetCluster,
            ext_management_system: { name: 'some provider' },
            [V2V_TARGET_PROVIDER_NETWORK_KEYS[targetProvider]]: [{}]
          }
        }
      });

      return store.dispatch(fetchTargetNetworksAction('/api', '1', targetProvider)).then(() => {
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
      });
    });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: '/api',
      status: 404,
      response: { error: 'error' }
    });

    return store.dispatch(fetchTargetNetworksAction('/api', '1', 'provider')).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});

describe('fetchPublicCloudNetworksAction', () => {
  const targetTenant = cloudTenants.resources[0];

  test('dispatches PENDING and FULFILLED actions and sets the clusterId and providerName', () => {
    mockRequest({
      method: 'GET',
      url: '/api',
      status: 200,
      response: { data: { resources: [{}] } }
    });

    return store.dispatch(fetchPublicCloudNetworksAction('/api', targetTenant)).then(() => {
      const actions = store.getActions();
      expect(actions).toMatchSnapshot();
    });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: '/api',
      status: 404,
      response: { error: 'error' }
    });

    return store.dispatch(fetchPublicCloudNetworksAction('/api', targetTenant)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
