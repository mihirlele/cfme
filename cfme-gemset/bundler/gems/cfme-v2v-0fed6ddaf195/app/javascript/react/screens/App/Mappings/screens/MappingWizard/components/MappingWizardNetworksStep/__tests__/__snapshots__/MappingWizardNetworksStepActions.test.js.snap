// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPublicCloudNetworksAction dispatches PENDING and FULFILLED actions and sets the clusterId and providerName 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_PUBLIC_CLOUD_NETWORKS_PENDING",
  },
  Object {
    "payload": Array [
      Object {
        "clusterId": "42000000000003",
        "providerName": "Brilliant OpenStack",
      },
    ],
    "type": "FETCH_V2V_PUBLIC_CLOUD_NETWORKS_FULFILLED",
  },
]
`;

exports[`fetchPublicCloudNetworksAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_PUBLIC_CLOUD_NETWORKS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_V2V_PUBLIC_CLOUD_NETWORKS_REJECTED",
  },
]
`;

exports[`fetchTargetNetworksAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TARGET_NETWORKS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_V2V_TARGET_NETWORKS_REJECTED",
  },
]
`;

exports[`fetchTargetNetworksAction if the target provider is OSP, dispatches PENDING and FULFILLED actions and sets the clusterId and providerName 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TARGET_NETWORKS_PENDING",
  },
  Object {
    "payload": Object {
      "targetNetworks": Array [
        Object {
          "clusterId": "42000000000003",
          "providerName": "some provider",
        },
      ],
    },
    "type": "FETCH_V2V_TARGET_NETWORKS_FULFILLED",
  },
]
`;

exports[`fetchTargetNetworksAction if the target provider is RHV, dispatches PENDING and FULFILLED actions and sets the clusterId and providerName 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TARGET_NETWORKS_PENDING",
  },
  Object {
    "payload": Object {
      "targetNetworks": Array [
        Object {
          "clusterId": "1",
          "providerName": "some provider",
        },
      ],
    },
    "type": "FETCH_V2V_TARGET_NETWORKS_FULFILLED",
  },
]
`;
